auth/certificate_authenticator.cc

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ *
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "auth/certificate_authenticator.hh"
+
+#include <regex>
+
+#include "utils/class_registrator.hh"
+#include "replica/database.hh"
+#include "cql3/query_processor.hh"
+#include "db/config.hh"
+
+static const auto CERT_AUTH_NAME = "com.scylladb.auth.CertificateAuthenticator";
+const std::string_view auth::certificate_authenticator_name(CERT_AUTH_NAME);
+
+static logging::logger clogger("certificate_authenticator");
+
+static const std::string cfg_source_attr = "source";
+static const std::string cfg_query_attr = "query";
+
+static const std::string cfg_source_subject = "SUBJECT";
+static const std::string cfg_source_altname = "ALTNAME";
+
+static const class_registrator<auth::authenticator
+    , auth::certificate_authenticator
+    , cql3::query_processor&
+    , ::service::migration_manager&> cert_auth_reg(CERT_AUTH_NAME);
+
+enum class auth::certificate_authenticator::query_source {
+    subject, altname
+};
+
+auth::certificate_authenticator::certificate_authenticator(cql3::query_processor& qp, ::service::migration_manager&)
+    : _queries([&] {
+        auto& conf = qp.db().get_config();
+        auto queries = conf.auth_certificate_role_queries();
+
+        if (queries.empty()) {
+            throw std::invalid_argument("No role extraction queries specified.");
+        }
+
+        std::vector<std::pair<query_source, std::regex>> res;
+
+        for (auto& map : queries) {
+            // first, check for any invalid config keys
+            if (map.size() == 2) {
+                try {
+                    auto& source = map.at(cfg_source_attr);
+                    std::string query = map.at(cfg_query_attr);
+
+                    std::transform(source.begin(), source.end(), source.begin(), ::toupper);
+
+                    std::regex ex(query);
```

Comment:
We stopped using std::regex due to stalls.

Please switch to boost::regex, and pre-compute the regex so it isn't compiled on every connection.

Reply:
This is the pre-computation. It is literally the loop in constructor transforming config values into runtime matchers. 
Honestly, it seems unlikely also given the kind of regex one expects here, that std:: vs boost:: would yield any measurable difference, but will switch...